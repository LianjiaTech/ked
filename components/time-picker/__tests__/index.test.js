import React from 'react';
import { mount, render } from 'enzyme';
import RcTimePicker from 'rc-time-picker/lib/TimePicker';
import moment from 'moment';
import TimePicker from '..';
import focusTest from '../../../tests/shared/focusTest';
import mountTest from '../../../tests/shared/mountTest';

describe('TimePicker', () => {
  const errorSpy = jest.spyOn(console, 'error').mockImplementation(() => {});

  afterEach(() => {
    errorSpy.mockReset();
  });

  afterAll(() => {
    errorSpy.mockRestore();
  });

  focusTest(TimePicker);
  mountTest(TimePicker);

  it('renders addon correctly', () => {
    const addon = () => <button type="button">Ok</button>;
    const wrapper = mount(<TimePicker addon={addon} />);
    const rcTimePicker = wrapper.find(RcTimePicker);
    const addonWrapper = render(rcTimePicker.props().addon());

    expect(addonWrapper).toMatchSnapshot();
  });

  it('allowEmpty deprecated', () => {
    mount(<TimePicker allowEmpty />);
    expect(errorSpy).toHaveBeenCalledWith(
      'Warning: [antd: TimePicker] `allowEmpty` is deprecated. Please use `allowClear` instead.',
    );
  });

  it('not render clean icon when allowClear is false', () => {
    const wrapper = mount(
      <TimePicker defaultValue={moment('2000-01-01 00:00:00')} allowClear={false} />,
    );
    expect(wrapper.render()).toMatchSnapshot();
  });

  it('handleChange should work correctly', done => {
    const date = moment('2000-01-01 00:00:00');
    const onChange = (value, formattedValue) => {
      expect(value).toBe(date);
      expect(formattedValue).toBe(date.format('HH:mm:ss'));
      done();
    };
    const wrapper = mount(<TimePicker onChange={onChange} />).instance();
    wrapper.handleChange(date);
  });

  it('handleOpenClose should work correctly', done => {
    const onOpenChange = open => {
      expect(open).toBe(true);
      done();
    };
    const wrapper = mount(<TimePicker onOpenChange={onOpenChange} />).instance();
    wrapper.handleOpenClose({ open: true });
  });

  it('clearIcon should render correctly', () => {
    const clearIcon = <div className="test-clear-icon">test</div>;
    const wrapper = mount(<TimePicker clearIcon={clearIcon} />);
    expect(wrapper.find('Picker').prop('clearIcon')).toEqual(
      <div className="test-clear-icon ant-time-picker-clear">test</div>,
    );
  });

  it('prop locale should works', () => {
    const locale = {
      placeholder: 'Избери дата',
    };
    const wrapper = mount(
      <TimePicker defaultValue={moment('2000-01-01 00:00:00')} open locale={locale} />,
    );
    expect(wrapper.render()).toMatchSnapshot();
  });
});
