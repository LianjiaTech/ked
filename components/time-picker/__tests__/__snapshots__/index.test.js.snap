// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimePicker not render clean icon when allowClear is false 1`] = `
<span
  class="ant-time-picker"
>
  <input
    class="ant-time-picker-input"
    placeholder="Select time"
    type="text"
    value="00:00:00"
  />
  <span
    class="ant-time-picker-icon"
  >
    <i
      aria-label="icon: clock-circle"
      class="anticon anticon-clock-circle ant-time-picker-clock-icon"
    >
      <svg
        aria-hidden="true"
        class=""
        data-icon="clock-circle"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <path
          d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
        />
        <path
          d="M686.7 638.6L544.1 535.5V288c0-4.4-3.6-8-8-8H488c-4.4 0-8 3.6-8 8v275.4c0 2.6 1.2 5 3.3 6.5l165.4 120.6c3.6 2.6 8.6 1.8 11.2-1.7l28.6-39c2.6-3.7 1.8-8.7-1.8-11.2z"
        />
      </svg>
    </i>
  </span>
</span>
`;

exports[`TimePicker prop locale should works 1`] = `
Array [
  <span
    class="ant-time-picker"
    style=""
  >
    <input
      class="ant-time-picker-input"
      placeholder="Избери дата"
      type="text"
      value="00:00:00"
    />
    <span
      class="ant-time-picker-icon"
    >
      <i
        aria-label="icon: clock-circle"
        class="anticon anticon-clock-circle ant-time-picker-clock-icon"
      >
        <svg
          aria-hidden="true"
          class=""
          data-icon="clock-circle"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
          />
          <path
            d="M686.7 638.6L544.1 535.5V288c0-4.4-3.6-8-8-8H488c-4.4 0-8 3.6-8 8v275.4c0 2.6 1.2 5 3.3 6.5l165.4 120.6c3.6 2.6 8.6 1.8 11.2-1.7l28.6-39c2.6-3.7 1.8-8.7-1.8-11.2z"
          />
        </svg>
      </i>
    </span>
    <i
      aria-label="icon: close-circle"
      class="anticon anticon-close-circle ant-time-picker-clear"
      tabindex="-1"
    >
      <svg
        aria-hidden="true"
        class=""
        data-icon="close-circle"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <path
          d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm165.4 618.2l-66-.3L512 563.4l-99.3 118.4-66.1.3c-4.4 0-8-3.5-8-8 0-1.9.7-3.7 1.9-5.2l130.1-155L340.5 359a8.32 8.32 0 0 1-1.9-5.2c0-4.4 3.6-8 8-8l66.1.3L512 464.6l99.3-118.4 66-.3c4.4 0 8 3.5 8 8 0 1.9-.7 3.7-1.9 5.2L553.5 514l130 155c1.2 1.5 1.9 3.3 1.9 5.2 0 4.4-3.6 8-8 8z"
        />
      </svg>
    </i>
  </span>,
  <div>
    <div
      class="ant-time-picker-panel ant-time-picker-panel-column-3 ant-time-picker-panel-placement-bottomLeft slide-up-appear"
      style="left: -999px; top: -1001px;"
    >
      <div
        class="ant-time-picker-panel-inner"
      >
        <div
          class="ant-time-picker-panel-input-wrap"
        >
          <input
            class="ant-time-picker-panel-input"
            placeholder="Избери дата"
            value="00:00:00"
          />
        </div>
        <div
          class="ant-time-picker-panel-combobox"
        >
          <div
            class="ant-time-picker-panel-select"
          >
            <ul>
              <li
                class="ant-time-picker-panel-select-option-selected"
                role="button"
                tabindex="0"
              >
                00
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                01
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                02
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                03
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                04
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                05
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                06
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                07
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                08
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                09
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                10
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                11
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                12
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                13
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                14
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                15
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                16
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                17
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                18
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                19
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                20
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                21
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                22
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                23
              </li>
            </ul>
          </div>
          <div
            class="ant-time-picker-panel-select"
          >
            <ul>
              <li
                class="ant-time-picker-panel-select-option-selected"
                role="button"
                tabindex="0"
              >
                00
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                01
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                02
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                03
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                04
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                05
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                06
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                07
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                08
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                09
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                10
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                11
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                12
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                13
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                14
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                15
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                16
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                17
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                18
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                19
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                20
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                21
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                22
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                23
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                24
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                25
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                26
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                27
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                28
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                29
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                30
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                31
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                32
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                33
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                34
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                35
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                36
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                37
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                38
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                39
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                40
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                41
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                42
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                43
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                44
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                45
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                46
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                47
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                48
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                49
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                50
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                51
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                52
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                53
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                54
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                55
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                56
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                57
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                58
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                59
              </li>
            </ul>
          </div>
          <div
            class="ant-time-picker-panel-select"
          >
            <ul>
              <li
                class="ant-time-picker-panel-select-option-selected"
                role="button"
                tabindex="0"
              >
                00
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                01
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                02
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                03
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                04
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                05
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                06
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                07
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                08
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                09
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                10
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                11
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                12
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                13
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                14
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                15
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                16
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                17
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                18
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                19
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                20
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                21
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                22
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                23
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                24
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                25
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                26
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                27
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                28
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                29
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                30
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                31
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                32
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                33
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                34
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                35
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                36
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                37
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                38
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                39
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                40
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                41
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                42
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                43
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                44
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                45
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                46
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                47
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                48
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                49
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                50
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                51
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                52
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                53
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                54
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                55
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                56
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                57
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                58
              </li>
              <li
                class=""
                role="button"
                tabindex="0"
              >
                59
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;

exports[`TimePicker renders addon correctly 1`] = `
<div
  class="ant-time-picker-panel-addon"
>
  <button
    type="button"
  >
    Ok
  </button>
</div>
`;
